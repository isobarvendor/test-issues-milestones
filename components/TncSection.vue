<template>
<div class="tnc-component" v-if="!data">
    <div class="tnc-section" >
        <div class="primary-section">
            <div class="header">Terms & Conditions
                <p> last updated: 30 July 2020</p>
            </div>
            <div class="desc">
                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                    Pellentesque scelerisque laoreet sem a tempor.
                    Vivamus rhoncus nibh id sagittis vulputate. Nulla sit amet lectus erat. Sed luctus,
                    massa dictum sagittis consectetur, diam lorem placerat nibh, quis suscipit diam lorem at libero.
                    Aliquam erat volutpat. Etiam ante nisi, consequat aliquam est vitae, pulvinar euismod neque. Nunc ultricies
                    egestas odio, nec congue augue fermentum nec. Mauris ac quam quis justo varius faucibus.
                    Phasellus laoreet ac dui vitae maximus. Nunc id maximus nisl.
                </p>
                <div class="links">
                    <p><a href="#">Header 1</a></p>
                    <p><a href="#">Header 2</a></p>
                    <p><a href="#">Header 3</a></p>
                    <p><a href="#">Header 4</a></p>
                    <p><a href="#">Header 5</a></p>
                    <p><a href="#">Header 6</a></p>
                </div>
            </div>
        </div>
        <div class="logo">
            <img src="/img/icons/coke.png"/>
        </div>
    </div>
    <div class="secondary-section">
        <div class="secondary-component">
            <div class="sub-header">Header 1</div>
            <div class="desc">
                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                    Pellentesque scelerisque laoreet sem a tempor.
                    Vivamus rhoncus nibh id sagittis vulputate. Nulla sit amet lectus erat. Sed luctus,
                    massa dictum sagittis consectetur, diam lorem placerat nibh, quis suscipit diam lorem at libero.
                    Aliquam erat volutpat. Etiam ante nisi, consequat aliquam est vitae, pulvinar euismod neque. Nunc ultricies
                    egestas odio, nec congue augue fermentum nec. Mauris ac quam quis justo varius faucibus.
                    Phasellus laoreet ac dui vitae maximus. Nunc id maximus nisl. <a href="mailto:coke@sg.com"> coke@sg.com</a>
                </p>
            </div>
        </div>
    </div>
    <div class="secondary-section">
        <div class="secondary-component">
            <div class="sub-header">Header 2</div>
            <div class="desc">
                <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                    Pellentesque scelerisque laoreet sem a tempor.
                    Vivamus rhoncus nibh id sagittis vulputate. Nulla sit amet lectus erat. Sed luctus,
                    massa dictum sagittis consectetur, diam lorem placerat nibh, quis suscipit diam lorem at libero.
                    Aliquam erat volutpat. Etiam ante nisi, consequat aliquam est vitae, pulvinar euismod neque. Nunc ultricies
                    egestas odio, nec congue augue fermentum nec. Mauris ac quam quis justo varius faucibus.
                </p>
            </div>
        </div>
    </div>
</div>
<div class="tnc-component" v-else>
    <div class="tnc-section" >
        <div class="primary-section">
            <div class="header">{{tnc.TncTop[0].Title}}
                <p> {{tnc.TncTop[0].LastUpdate}}</p>
            </div>
            <div class="desc">
                <p>{{tnc.TncTop[0].Description}}
                </p>
                <div class="links">
                    <p v-for="(item,index) in tnc.TncBottom"><a href="#">{{item.Title}}</a></p>
                </div>
            </div>
        </div>
        <div class="logo">
            <img src="/img/icons/coke.png"/>
        </div>
    </div>
    <div class="secondary-section" v-for="(item,index) in tnc.TncBottom">
        <div class="secondary-component">
            <div class="sub-header">{{item.Title}}</div>
            <div class="desc">
                <p>{{item.Description}}
                </p>
            </div>
        </div>
    </div>
</div>
</template>

<script>
export default {
  name: "TncSection",
  components: {
  },
  data() {
    return {
      currentPage: 1
    };
  },
  props: {
    data: null,
  },
  computed:{
    tnc(){
      return this.data
    }
  },
  methods:{
  },
  beforeMount() {},
  mounted(){

  }
};
</script>

<style lang="scss">
.tnc-component{
    .tnc-section{
        display: flex;

        .primary-section{
            display: flex;
            padding-bottom: 50px;
            border-bottom: 1px solid #585858;
            .header{
                width: 40%;
                p{
                    font-size: 16px;
                    margin-top: 35px;
                }
            }
            .desc{
                width: 50%;
                p{
                    line-height: 24px;
                }
                .links{
                    margin-top: 40px;
                    p{
                        margin-bottom: 10px;
                    }
                }
            }
        }
         .logo{
            align-self: flex-end;
            margin-left: 35px;
        }

    }

    .secondary-section{
        width: 78%;
        .secondary-component{
            margin-top: 40px;
            display:flex;
            padding-bottom: 40px;
            border-bottom: 1px solid #585858;
            .sub-header{
                font-size: 25px;
                width: 34%;
            }
            .desc{
                width: 50%;
                margin-left: 50px;
                p{
                    line-height: 24px;
                }
            }
        }
    }
    @media only screen and (max-width: 1199px) {
        .tnc-section{
            flex-direction: column-reverse;
            .primary-section{
                flex-direction: column;
                .header{
                    width:100%;
                    font-size: 18px;
                    margin-top:35px;
                    p{
                        margin-top:15px;
                        margin-left: 20px;
                    }
                }
                .desc{
                    margin-top:30px;
                    width: 100%;
                }
            }
            .logo{
                margin-left: 0px;
                align-self: center;
            }
        }
        .secondary-section{
            width: 100%;
            .secondary-component{
                .sub-header{
                    font-size: 18px;
                }
                flex-direction: column;
                .desc{
                    margin-left: 0px;
                    width: 100%;
                    margin-top: 30px;
                }
            }
        }
    }
}
</style>
