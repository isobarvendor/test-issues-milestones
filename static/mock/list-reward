[ {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   },
   {       "public_code": "shopee-test-2",
       "secret_code": "005ZWWZXFE82",
       "url": "https://barcode-stage.web.app/c/shopee-test-2",
        "secret_code_image": "/img/landing/barcode.png",
       "store_name": "Shopee",
       "coupon_expired_at": "2021-05-09 17:50:00",
       "creative_image": "https://cdn2.grivy.com/Testing%20Juice/JUICE_BANNER-1080x480.jpg"
   }
]
