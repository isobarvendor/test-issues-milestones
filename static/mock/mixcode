{
                      "programId": 453269,
                      "code": "3WTD90GE5",
                      "lot": {
                        "lotId": 4455843436,
                        "lotName": "Coke 500ml PET",
                        "activateDate": 1611208800000,
                        "inactivateDate": 1627603200000,
                        "active": true,
                        "expired": false,
                        "bevProdPkg": {
                          "trademarkCd": "05",
                          "trademarkName": "Coca-Cola",
                          "brandCd": "001",
                          "brandName": "Coca-Cola",
                          "bevProdCd": "0001",
                          "bevProdName": "Coca-Cola",
                          "caffeinated": true,
                          "carbonated": true,
                          "calorieCategory": {
                            "code": 1,
                            "description": "Regular"
                          }
                        },
                        "releaseId": 494494,
                        "releaseName": "Coke PET",
                        "organizationId": 157168,
                        "organizationName": "Harish",
                        "realtimeCode": false,
                        "testRelease": false,
                        "pointValue": 0,
                        "bevProdOptions": []
                      },
                      "redeemed": false,
                      "actualCode": "3WTD90GE5"
                    }
