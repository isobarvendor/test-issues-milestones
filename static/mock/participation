{"id":16,"questionId":1,"answerId":3,"participationId":17}
