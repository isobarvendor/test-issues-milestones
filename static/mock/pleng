{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p>3 months subscription to Pleng Pro</p>",
      "shortDescription": "<p>ឈ្នះថ្លៃសេវា Pleng Pro Subscription រយៈពេល 3 ខែ</p>",
      "priority": 1,
      "prizeId": "ks8o4ixh",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8o4ixh/CAM prizes — Pleng Pro@2x.png",
      "name": "3 months subscription to Pleng Pro ",
      "redemptionLink": "https://pleng.onelink.me/pp89?pid=ddlink&af_web_dp=https%3A%2F%2Fwww.smart.com.kh%2Fkm%2Fpleng%2F&af_dp=plengmusic%3A%2F% 2F&af_partner_name=cc",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះ Pleng Pro Subscription រយៈពេល ៣ ខែ! រីករាយនូវការស្តាប់ចម្រៀងឥតដែនកំណត់។ សូមដោនឡូតកម្មវិធី Pleng ពី Smart ដើម្បីបញ្ចូលលេខកូដ និងដាក់ដំណើរការ Smart Pleng Pro។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ ២០២១។</p>",
      "voucherCode": "CNXFUUAKAY"
      
    }
  }
}