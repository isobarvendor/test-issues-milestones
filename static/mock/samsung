{
  "burnResult": [
    {
      "pincode": "0J6TC2VP5",
      "programId": "453269",
      "lotId": "445580",
      "burned": true
    }
  ],
  "participationId": "1ylazdy9kkxmpa53",
  "instantWinResult": {
    "winner": true,
    "redeemedPrize": {
      "barcodeType": 1,
      "deliveryType": 1,
      "active": true,
      "description": "<p></p>",
      "shortDescription": "<p>ទូរស័ព្ទសាមសុង Galaxy Note20 Ultra</p>",
      "priority": 1,
      "prizeId": "ks8ns5hv",
      "imgUrl": "https://d1tnwdlrciourl.cloudfront.net/prizeImages/z2xtqi9ks8ndwx3/ks8ns5hv/1.CAM prize-N20 Ultra@2x.png",
      "totalAmount": 20,
      "minAge": 13,
      "name": "Samsung Galaxy Note20 Ultra",
      "redemptionLink": "",
      "configurationId": "z2xtqi9ks8ndwx3",
      "redeemDescription": "<p>អបអរសាទរ! អ្នកបានឈ្នះទូរស័ព្ទសាមសុង Galaxy Note20 Ultra! សូមទាក់ទងទៅលេខ 1-800-20-2222 (មិនគិតថ្លៃសេវាទូរស័ព្ទ ២៤/៧) ដើម្បីដឹងពីរបៀបប្តូរយករង្វាន់។ សូមប្តូរយករង្វាន់ឲ្យបានមុនថ្ងៃទី ៣០ ខែ វិច្ឆិកា ឆ្នាំ</p>",
      "voucherCode": "CNXFUUAKAY"
    }
  }
}