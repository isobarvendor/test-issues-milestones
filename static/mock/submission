{
            "burnResult": [
                {
                    "pincode": "0J6TC2VP5",
                    "programId": "453269",
                    "lotId": "445580",
                    "burned": true
                }
            ],
            "participationId": "1ylazdy9kkxmpa53",
            "instantWinResult": {
                "winner": true,
              "redeemedPrize": {
                    "prizeId": "kkqhy9so",
                    "voucherCode": "Voucher891",
                    "status": "claimed",
                    "expiryDate": 1620018590000,
                    "name": "JOOX VIP PASS",
                    "description":"Conglratulations",
                    "shortDescription": "http://google.com",
                    "redeemDescription": "http://abcd",
                    "imgUrl": "/img/landing/week 1 prize.png",
                    "barcodeType": 1,
                    "emailSent": false,
                    "emailMessage": "Email could not be sent"
                }
            }
        }
