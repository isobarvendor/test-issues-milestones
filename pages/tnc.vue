<template>
  <div v-if="data" id="main" class="tnc">
    <div class="wrapper">
      <div class="container">
        <TncSection :data="data[0].Tnc"/>
      </div>

    </div>
  </div>

</template>

<script>
import TncSection from '@/components/TncSection'
export default {
  name:"Tnc",
  components:{
    TncSection
  },
  data(){
    return{
      dataStatus:{},

    }
  },
  head() {
    return {
      title: "Coke Campaign Title",
      meta: [
        {
          hid: "description",
          name: "description",
          content: "Home page description"
        }
      ],
      noscript: [],
      /*script: [
        {
          hid: "anime",
          src: "/js/anime.min.js"
        }
      ],*/
      css: []
    };
  },
 computed: {
    data(){
        return this.$store.getters.getCMSContent;
     }
  },

};
</script>

<style lang="scss">
@import '~assets/scss/variables.scss';
  html, body{
    background: $body-bg-color-wallet;
    @media only screen and (max-width: 767px) {
      .tnc{
      }
    }
  }
</style>
