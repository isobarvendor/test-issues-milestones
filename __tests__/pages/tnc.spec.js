import { mount,  createLocalVue } from '@vue/test-utils';
import Vuex from 'vuex';
import VueRouter from 'vue-router'

import tnc from '@/pages/tnc.vue';

import TncSection from '@/components/TncSection'


import {mockCMSConfig,mockCMSContent} from "@/__mock__/mock";





let localVue, store,router;


  beforeEach(() => {
    localVue = createLocalVue();
    localVue.use(Vuex);
    localVue.use(VueRouter);
    router = new VueRouter();
    store = new Vuex.Store({
      getters: {
        getCMSContent() {
          return mockCMSContent;
        },
        getCMSConfig(){
          return mockCMSConfig[0];
        }
      },
      state:{
        token:"",
        language: "en",
        listWallet:{
          walletStatus:[
            {amount :30}
          ]
        },

      }
    });
  });

  const factory = () => {
    return mount(tnc, {

      localVue, store,router
    });
  };



describe('tnc', () => {


  test('renders a TncSection', () => {
    const wrapper = factory();
    const component= wrapper.findComponent(TncSection);
    expect(component.exists()).toBeTruthy();
  });


  test("mounts properly", () => {
    const wrapper = factory();
    expect(wrapper.vm).toBeTruthy();
  });

  test("renders properly", () => {
    const wrapper = factory();
    expect(wrapper.html()).toMatchSnapshot();
  });

});
