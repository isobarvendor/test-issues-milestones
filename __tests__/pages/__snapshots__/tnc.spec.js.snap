// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`tnc renders properly 1`] = `
<div id="main" class="tnc">
  <div class="wrapper">
    <div class="container">
      <div class="tnc-component">
        <div class="tnc-section">
          <div class="primary-section">
            <div class="header">Terms and Conditions
              <p> Last update :10 july 2020</p>
            </div>
            <div class="desc">
              <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                Pellentesque scelerisque laoreet sem a tempor.
                Vivamus rhoncus nibh id sagittis vulputate. Nulla sit amet lectus erat. Sed luctus,
                massa dictum sagittis consectetur, diam lorem placerat nibh, quis suscipit diam lorem at libero.
                Aliquam erat volutpat. Etiam ante nisi, consequat aliquam est vitae, pulvinar euismod neque. Nunc ultricies
                egestas odio, nec congue augue fermentum nec. Mauris ac quam quis justo varius faucibus.
                Phasellus laoreet ac dui vitae maximus. Nunc id maximus nisl.

              </p>
              <div class="links">
                <p><a href="#">Header1</a></p>
                <p><a href="#">Header2</a></p>
                <p><a href="#">Header 3</a></p>
              </div>
            </div>
          </div>
          <div class="logo"><img src="/img/icons/coke.png"></div>
        </div>
        <div class="secondary-section">
          <div class="secondary-component">
            <div class="sub-header">Header1</div>
            <div class="desc">
              <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                Pellentesque scelerisque laoreet sem a tempor.
                Vivamus rhoncus nibh id sagittis vulputate. Nulla sit amet lectus erat. Sed luctus,
                massa dictum sagittis consectetur, diam lorem placerat nibh, quis suscipit diam lorem at libero.
                Aliquam erat volutpat. Etiam ante nisi, consequat aliquam est vitae, pulvinar euismod neque. Nunc ultricies
                egestas odio, nec congue augue fermentum nec. Mauris ac quam quis justo varius faucibus.
                Phasellus laoreet ac dui vitae maximus. Nunc id maximus nisl.

              </p>
            </div>
          </div>
        </div>
        <div class="secondary-section">
          <div class="secondary-component">
            <div class="sub-header">Header2</div>
            <div class="desc">
              <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                Pellentesque scelerisque laoreet sem a tempor.
                Vivamus rhoncus nibh id sagittis vulputate. Nulla sit amet lectus erat. Sed luctus,
                massa dictum sagittis consectetur, diam lorem placerat nibh, quis suscipit diam lorem at libero.
                Aliquam erat volutpat. Etiam ante nisi, consequat aliquam est vitae, pulvinar euismod neque. Nunc ultricies
                egestas odio, nec congue augue fermentum nec. Mauris ac quam quis justo varius faucibus.
                Phasellus laoreet ac dui vitae maximus. Nunc id maximus nisl.

              </p>
            </div>
          </div>
        </div>
        <div class="secondary-section">
          <div class="secondary-component">
            <div class="sub-header">Header 3</div>
            <div class="desc">
              <p>Lorem ipsum dolor sit amet, consectetur adipiscing elit.
                Pellentesque scelerisque laoreet sem a tempor.
                Vivamus rhoncus nibh id sagittis vulputate. Nulla sit amet lectus erat. Sed luctus,
                massa dictum sagittis consectetur, diam lorem placerat nibh, quis suscipit diam lorem at libero.
                Aliquam erat volutpat. Etiam ante nisi, consequat aliquam est vitae, pulvinar euismod neque. Nunc ultricies
                egestas odio, nec congue augue fermentum nec. Mauris ac quam quis justo varius faucibus.
                Phasellus laoreet ac dui vitae maximus. Nunc id maximus nisl.

              </p>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
